thunder/dynamo/compiler.py

```diff
@@ -0,0 +1,62 @@
+import torch
+
+from collections import OrderedDict
+
+
+class ThunderCompiler:
+    def __init__(self, **thunder_options):
+        """
+        A class that compiles a `fx.GraphModule` to a `thunder.ThunderModule`.
+        This class is meant to be used as a backend for the `torch.compile`
+        function.
+
+        Keyword arguments:
+            thunder_options: a dictionary of options to pass to `thunder.jit`.
+
+        Example:
+            >>> import torch
+            >>> from thunder.dynamo import ThunderCompiler
+            >>> backend = ThunderCompiler()
+            >>> x = torch.ones(2, requires_grad=True)
+            >>> @torch.compile(backend=backend)
+            ... def func(x):
+            ...     x = torch.sin(x)
+            ...     if x.sum() > 0:
+            ...         return x + 1
+            ...     else:
+            ...         return x - 1
+            >>> out = func(x)
+        """
+        from thunder import ThunderModule
+
+        # Thunder-compiled functions should be readily available for inspection
+        # and testing, so we will store them in a list. The order of the
+        # functions in the list will be the same as the order in which they were
+        # compiled. In addition, we will store a mapping from the ThunderModule
+        # to the GraphModule that was passed to ThunderCompiler. This will allow
+        # us to inspect the GraphModule that was compiled by Thunder.
+        self.thunder_fns: list[ThunderModule] = []
+        self.thunder_to_gm: OrderedDict[ThunderModule, torch.fx.GraphModule] = OrderedDict()
```

Comment:
[question] iirc `dict` reserves the insertion order, so I'm wondering what/which features of `OrderedDict` are needed